sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -258,7 +266,11 @@ object HiveThriftServer2 extends Logging {
 
     def onOperationClosed(id: String): Unit = synchronized {
       executionList(id).closeTimestamp = System.currentTimeMillis
+      val lastState = executionList(id).state
       executionList(id).state = ExecutionState.CLOSED
+      if (lastState == ExecutionState.STARTED || lastState == ExecutionState.COMPILED) {
+        totalRunning -= 1
+      }
```

Comment:
does FINISHED need to be handled somewhere?

Reply:
> does FINISHED need to be handled somewhere?

this place is to handle situation like : we have call close before finish, then the totalRunning value should -1 too. If we call `cancel()` or it finished, then when we call closeOperation, we won't do `totalRunning -= 1`. Since for FINISHED, has do `totalRunning -= 1 `, so here don't need to contain FINISHED